service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EncounterComponentsMapperTest.java

```diff
@@ -202,4 +205,9 @@ private Encounter extractEncounter(Bundle bundle) {
             Arguments.of(INPUT_BUNDLE_WITH_LIST_NOT_IN_TOPIC_OR_CATEGORY)
         );
     }
+
+    private void setUpMock() {
```

Comment:
You could consider using a `@NestedClass` to group tests by mocking requirements rather than repeating this method invocation. Is there a common category for all these tests that need `setUpMock()`?

Reply:
the name for the method is wrong, I think it should be rather initializeMessageContext because it requires message context to be set up with different input bundle for each test. I will change it a little, I am not sure how to use nested class in this case. Let me know what you think or how you would like it to look.

Reply:
With your latest changes, a `@NestedClass` would no longer work. 👍